src/index.html

```diff
@@ -2,45 +2,53 @@
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <meta
-      name="viewport"
-      content="width=device-width, initial-scale=1.0"
-    />
-    <title>Promise function</title>
-    <link
-      rel="stylesheet"
-      href="./styles/main.scss"
-    />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Document</title>
   </head>
   <body>
-    <form>
-      <label for="login">
-        Login
-        <input
-          name="login"
-          type="text"
-          placeholder="Login"
-          id="login"
-        />
-      </label>
-      <label for="password">
-        Password
-        <input
-          name="password"
-          type="password"
-          placeholder="Password"
-          id="password"
-        />
-      </label>
+    <script>
+      function waitFor(element, eventName) {
+        return new Promise((resolve) => {
+          const handler = (event) => {
+            if (event.target === element) {
+              element.removeEventListener(eventName, handler);
+              resolve(`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`);
+            }
+          };
 
-      <button
-        type="button"
-        id="submit"
-      >
-        Login
-      </button>
-    </form>
+          // Check if the event has already happened
+          if (eventName === 'DOMContentLoaded' && document.readyState === 'complete') {
+            resolve(`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`);
+          } else {
+            element.addEventListener(eventName, handler);
+          }
+        });
+      }
 
-    <script src="scripts/main.js"></script>
+      function printMessage(message) {
+        const div = document.createElement('div');
+        div.className = 'message';
+        div.textContent = message;
+        document.body.appendChild(div);
+      }
+
+      // Example usage:
+      document.addEventListener('DOMContentLoaded', () => {
+        const loginInput = document.getElementById('login');
+        const passwordInput = document.getElementById('password');
+        const submitButton = document.getElementById('submit');
+
+        const elements = [loginInput, passwordInput, submitButton];
+        const events = ['click', 'blur'];
+
+        events.forEach((eventName) => {
+          document.body.addEventListener(eventName, (event) => {
+            if (elements.includes(event.target)) {
+              waitFor(event.target, eventName).then(printMessage);
+            }
+          });
+        });
+      });
+    </script>
```

Comment:
It's generally a good practice to separate JavaScript code from HTML. Consider moving this script to an external JavaScript file and then include it in your HTML using the 'script' tag with the 'src' attribute.